compiler/src/dmd/lexer.d

```diff
@@ -1555,7 +1555,10 @@ class Lexer
                     }
                     if (ndigits != 2 && !utf_isValidDchar(v))
                     {
-                        error(loc, "invalid UTF character \\U%08x", v);
+                        if (v >= 0xD800 && v <= 0xDFFF)
+                            error(loc, "the invalid UTF character \\U%08x is a UTF-16 surrogate, is your escape a UTF-16 code unit instead of a code point?", v);
```

Comment:
```
error(loc, "invalid UTF character \\U%08x is a UTF-16 surrogate");
errorSupplimental("escape sequences must be UTF-16 code code points");
```

Reply:
Okay, I've moved the surrogate message into the supplemental call.